// location-hide/index.js
module.exports = function() {
//  *██████╗██████╗*███████╗*█████╗*████████╗███████╗██████╗*****██████╗*██╗***██╗*******************************************************************************************
//  ██╔════╝██╔══██╗██╔════╝██╔══██╗╚══██╔══╝██╔════╝██╔══██╗****██╔══██╗╚██╗*██╔╝*******************************************************************************************
//  ██║*****██████╔╝█████╗**███████║***██║***█████╗**██║**██║****██████╔╝*╚████╔╝********************************************************************************************
//  ██║*****██╔══██╗██╔══╝**██╔══██║***██║***██╔══╝**██║**██║****██╔══██╗**╚██╔╝*********************************************************************************************
//  ╚██████╗██║**██║███████╗██║**██║***██║***███████╗██████╔╝****██████╔╝***██║**********************************************************************************************
//  *╚═════╝╚═╝**╚═╝╚══════╝╚═╝**╚═╝***╚═╝***╚══════╝╚═════╝*****╚═════╝****╚═╝**********************************************************************************************
//  *************************************************************************************************************************************************************************
//  ██╗****██╗██╗****██╗██╗****██╗***██╗**██╗*██████╗*██████╗*███╗***██╗██╗***██╗███████╗*█████╗*███╗***███╗██╗██╗**██╗***██╗*██████╗*███╗***██╗██╗*****██╗███╗***██╗███████╗
//  ██║****██║██║****██║██║****██║***██║**██║██╔═══██╗██╔══██╗████╗**██║╚██╗*██╔╝██╔════╝██╔══██╗████╗*████║██║██║**╚██╗*██╔╝██╔═══██╗████╗**██║██║*****██║████╗**██║██╔════╝
//  ██║*█╗*██║██║*█╗*██║██║*█╗*██║***███████║██║***██║██████╔╝██╔██╗*██║*╚████╔╝*█████╗**███████║██╔████╔██║██║██║***╚████╔╝*██║***██║██╔██╗*██║██║*****██║██╔██╗*██║█████╗**
//  ██║███╗██║██║███╗██║██║███╗██║***██╔══██║██║***██║██╔══██╗██║╚██╗██║**╚██╔╝**██╔══╝**██╔══██║██║╚██╔╝██║██║██║****╚██╔╝**██║***██║██║╚██╗██║██║*****██║██║╚██╗██║██╔══╝**
//  ╚███╔███╔╝╚███╔███╔╝╚███╔███╔╝██╗██║**██║╚██████╔╝██║**██║██║*╚████║***██║***██║*****██║**██║██║*╚═╝*██║██║███████╗██║██╗╚██████╔╝██║*╚████║███████╗██║██║*╚████║███████╗
//  *╚══╝╚══╝**╚══╝╚══╝**╚══╝╚══╝*╚═╝╚═╝**╚═╝*╚═════╝*╚═╝**╚═╝╚═╝**╚═══╝***╚═╝***╚═╝*****╚═╝**╚═╝╚═╝*****╚═╝╚═╝╚══════╝╚═╝╚═╝*╚═════╝*╚═╝**╚═══╝╚══════╝╚═╝╚═╝**╚═══╝╚══════╝
//  *************************************************************************************************************************************************************************

//  THIS SCRIPT CAN BE USED FOR PERSONAL OR COMMERCIAL USE (CC0). You can edit this srcipt or use it on your projects. You don´t need to backlink. If you write your own 
//  srcipts with this srcipt then please make a backlink to the producer page www.hornyfamily.online by placing this url anywhere in the srcipt file. Thank you

//  ASCII GENERATOR BY - www.patorjk.com



//  *█████╗*██████╗*██████╗**█████╗*██╗***██╗███████╗********██╗*****██████╗*██████╗******██╗███████╗*██████╗████████╗███████╗
//  ██╔══██╗██╔══██╗██╔══██╗██╔══██╗╚██╗*██╔╝██╔════╝*******██╔╝****██╔═══██╗██╔══██╗*****██║██╔════╝██╔════╝╚══██╔══╝██╔════╝
//  ███████║██████╔╝██████╔╝███████║*╚████╔╝*███████╗******██╔╝*****██║***██║██████╔╝*****██║█████╗**██║********██║***███████╗
//  ██╔══██║██╔══██╗██╔══██╗██╔══██║**╚██╔╝**╚════██║*****██╔╝******██║***██║██╔══██╗██***██║██╔══╝**██║********██║***╚════██║
//  ██║**██║██║**██║██║**██║██║**██║***██║***███████║****██╔╝*******╚██████╔╝██████╔╝╚█████╔╝███████╗╚██████╗***██║***███████║
//  ╚═╝**╚═╝╚═╝**╚═╝╚═╝**╚═╝╚═╝**╚═╝***╚═╝***╚══════╝****╚═╝*********╚═════╝*╚═════╝**╚════╝*╚══════╝*╚═════╝***╚═╝***╚══════╝


var _ = {}
var __ = []
var __src = []
var __content = []
var __id = []
var __class = []
var __1 = []
var __2 = []
var __3 = []
var __4 = []
var __5 = []





//  *██████╗*██████╗*███╗***██╗███████╗████████╗
//  ██╔════╝██╔═══██╗████╗**██║██╔════╝╚══██╔══╝
//  ██║*****██║***██║██╔██╗*██║███████╗***██║***
//  ██║*****██║***██║██║╚██╗██║╚════██║***██║***
//  ╚██████╗╚██████╔╝██║*╚████║███████║***██║***
//  *╚═════╝*╚═════╝*╚═╝**╚═══╝╚══════╝***╚═╝***

  const file_extension = filePath.split(/[.](\w[a-z0-9]+)+$/gmi);
        crypted_file = 'Location Hider by WWW.HORNYFAMILY.ONLINE -- CRYPTED FILE - ' + simpleTimestamp().replace(new RegExp(/[.]/gmi),"_");
        crypted_file3 = 'Location Hider by WWW.HORNYFAMILY.ONLINE -- PASTE THIS IN EXTERNAL SCRIPT AND INLCUDE IT TO YOUR WEBSITE' + simpleTimestamp().replace(new RegExp(/[.]/gmi),"_");
        crypted_file2 = crypted_file + '.' + file_extension[1];
        crypted_file4 = crypted_file3 + '.js'
        importedlooplength = Loop_Length_fake_include_scripts



      //  console.log('File Path:', filePath);
        console.log('File Extension of imported file:', file_extension[1]);
        console.log('Loop Length for fake include scripts:', importedlooplength);







//  ██╗███╗***███╗██████╗**██████╗*██████╗*████████╗****███████╗██╗██╗*****███████╗
//  ██║████╗*████║██╔══██╗██╔═══██╗██╔══██╗╚══██╔══╝****██╔════╝██║██║*****██╔════╝
//  ██║██╔████╔██║██████╔╝██║***██║██████╔╝***██║*******█████╗**██║██║*****█████╗**
//  ██║██║╚██╔╝██║██╔═══╝*██║***██║██╔══██╗***██║*******██╔══╝**██║██║*****██╔══╝**
//  ██║██║*╚═╝*██║██║*****╚██████╔╝██║**██║***██║*******██║*****██║███████╗███████╗
//  ╚═╝╚═╝*****╚═╝╚═╝******╚═════╝*╚═╝**╚═╝***╚═╝*******╚═╝*****╚═╝╚══════╝╚══════╝


fs.readFile(filePath, {encoding: 'utf-8'}, function(err,data){
    if (!err) {   

    _._ = data.toString();
    // console.log('received data: ' + _._); 







//   ███╗***███╗*█████╗*████████╗*██████╗██╗**██╗
//   ████╗*████║██╔══██╗╚══██╔══╝██╔════╝██║**██║
//   ██╔████╔██║███████║***██║***██║*****███████║
//   ██║╚██╔╝██║██╔══██║***██║***██║*****██╔══██║
//   ██║*╚═╝*██║██║**██║***██║***╚██████╗██║**██║
//   ╚═╝*****╚═╝╚═╝**╚═╝***╚═╝****╚═════╝╚═╝**╚═╝




    var regex = _._.match(/( href="([^"]*"))/gm)
    console.log('href matches - regex: ' + regex)

         if(regex){
         regex.forEach(function(prerun, key) {
         var pre_prerun = prerun.replace(new RegExp(/^ /gmi),"");
         __1.push(pre_prerun)
         }); // regex.forEach(function(prerun, key) {
         regex = __1
         } // if(regex){





    var regex_src = _._.match(/( src="([^"]*"))/gm) 
    console.log('src matches - regex_src: ' + regex_src)
         
         if(regex_src){
         regex_src.forEach(function(prerun, key) {
         var pre_prerun = prerun.replace(new RegExp(/^ /gmi),"");
         __2.push(pre_prerun)
         }); //  regex_src.forEach(function(prerun, key) {
         regex_src = __2
         } // if(regex_src){







    var regex_content = _._.match(/( content="([^"]*"))/gm) 
    console.log('content matches - regex_content: ' + regex_content)

         if(regex_content){
         regex_content.forEach(function(prerun, key) {
         var pre_prerun = prerun.replace(new RegExp(/^ /gmi),"");
         __3.push(pre_prerun)
         }); // regex_content.forEach(function(prerun, key) {
         regex_content = __3
         } // if(regex_content){







    var regex_id = _._.match(/( id="([^"]*"))/gm) 
    console.log('id matches - regex_id: ' + regex_id)

         if(regex_id){
         regex_id.forEach(function(prerun, key) {
         var pre_prerun = prerun.replace(new RegExp(/^ /gmi),"");
         __4.push(pre_prerun)
         }); // regex_id.forEach(function(prerun, key) {
         regex_id = __4
         } // if(regex_id){




    var regex_class = _._.match(/( class="([^"]*"))/gm) 
    console.log('class matches - regex_class: ' + regex_class)

         if(regex_class){
         regex_class.forEach(function(prerun, key) {
         var pre_prerun = prerun.replace(new RegExp(/^ /gmi),"");
         __5.push(pre_prerun)
         }); // regex_class.forEach(function(prerun, key) {
         regex_class = __5
         } // if(regex_class){









//  *██████╗██████╗*███████╗*█████╗*████████╗███████╗*********██╗███████╗****██╗███╗***██╗*██████╗██╗*****██╗***██╗██████╗*███████╗███████╗
//  ██╔════╝██╔══██╗██╔════╝██╔══██╗╚══██╔══╝██╔════╝*********██║██╔════╝****██║████╗**██║██╔════╝██║*****██║***██║██╔══██╗██╔════╝██╔════╝
//  ██║*****██████╔╝█████╗**███████║***██║***█████╗***********██║███████╗****██║██╔██╗*██║██║*****██║*****██║***██║██║**██║█████╗**███████╗
//  ██║*****██╔══██╗██╔══╝**██╔══██║***██║***██╔══╝******██***██║╚════██║****██║██║╚██╗██║██║*****██║*****██║***██║██║**██║██╔══╝**╚════██║
//  ╚██████╗██║**██║███████╗██║**██║***██║***███████╗****╚█████╔╝███████║****██║██║*╚████║╚██████╗███████╗╚██████╔╝██████╔╝███████╗███████║
//  *╚═════╝╚═╝**╚═╝╚══════╝╚═╝**╚═╝***╚═╝***╚══════╝*****╚════╝*╚══════╝****╚═╝╚═╝**╚═══╝*╚═════╝╚══════╝*╚═════╝*╚═════╝*╚══════╝╚══════╝



       /*  HREF */
       regex.forEach(function(href, key) {


        if (href.indexOf("<?php") >= 0) {
        // ..
        }
        else{

         if (href.indexOf("fonts.googleapis.com") >= 0) {
         // ..
         }
         else{


          if (href.indexOf("http") >= 0) {
         //******************************//

         var href_onlyurl = href.replace(new RegExp(/^ /gmi),"");
             href_onlyurl = href_onlyurl.replace(new RegExp(/href[=]["]/gmi),"");
             href_onlyurl = href_onlyurl.replace(new RegExp(/["]/gmi),"");

         //console.log('enter_1')
         var random = randomize('Aa', 100)
         var random2 = randomize('Aa', 100)
         var random3 = randomize('Aa', 100)
         var FINAL = "\nvar " + random + " = document.querySelectorAll('[data-" + random2 + "=\"" + random3 + "\"]');\n$('[data-" + random2 + "=\"" + random3 + "\"]').on('click', function(){\nlocation.href = \"" + href_onlyurl + "\";\n})\n"
         __.push(FINAL)

          if (_._.indexOf(href) >= 0) {
          //console.log('href found in _._');
          var replace = "data-" + random2 + "=\"" + random3 + "\""
          _._ = _._.replace(href,replace)
          } // if (_._.indexOf(href) >= 0) {
          //******************************//
          }
          else{
         //******************************//
         console.log('enter_href')
         var href_nolink_onlyurl = href.replace(new RegExp(/^ /gmi),"");
             href_nolink_onlyurl = href_nolink_onlyurl.replace(new RegExp(/href[=]["]/gmi),"");
             href_nolink_onlyurl = href_nolink_onlyurl.replace(new RegExp(/["]/gmi),"");


         var random = randomize('Aa', 100)
         var random2 = randomize('Aa', 100)
         var random3 = randomize('Aa', 100)
         var FINAL = "\nvar " + random + " = document.querySelectorAll('[data-" + random2 + "=\"" + random3 + "\"]');"
         + "\n$('[data-" + random2 + "=\"" + random3 + "\"]').attr(\"href\", \"" + href_nolink_onlyurl + "\");"
         __.push(FINAL)

          if (_._.indexOf(href) >= 0) {
          //console.log('href found in _._');
          var replace = "data-" + random2 + "=\"" + random3 + "\""
          _._ = _._.replace(href,replace)
          } // if (_._.indexOf(href) >= 0) {
          //******************************//


           } // else from    if (href.indexOf("http") >= 0) {
          } // else from  if (href.indexOf("<?php") >= 0) {

        } // else from if (href.indexOf("<?php") >= 0) {

      }); // regex.forEach(function(val, key) {






















         /*  src  */
         regex_src.forEach(function(src, key) {




        if (src.indexOf("______________.js") >= 0) {
        // ..
        }
        else{


         if (src.indexOf("jquery.min.js") >= 0) {
         // ..
         }
         else{

          if (src.indexOf("<?php") >= 0) {
          // ..
          }
          else{


           if ( src.match(/[.]js["]$/gmi) ) {

             //******************************//
         console.log('enter_src')
         var src_onlyurl = src.replace(new RegExp(/^ /gmi),"");
             src_onlyurl = src_onlyurl.replace(new RegExp(/src[=]["]/gmi),"");
             src_onlyurl = src_onlyurl.replace(new RegExp(/["]/gmi),"");


         var random = randomize('Aa', 100)
         var random2 = randomize('Aa', 100)
         var random3 = randomize('Aa', 100)
         var FINAL = "\nvar " + random + " = document.querySelectorAll('[data-" + random2 + "=\"" + random3 + "\"]');"
         + "\n$('[data-" + random2 + "=\"" + random3 + "\"]').attr(\"src\", \"" + src_onlyurl + "\");"
         + "\n$('[data-" + random2 + "=\"" + random3 + "\"]').attr(\"src\", \"" + "" + "\")"
         + ".delay( 10000 ); // IF YOU GOT LOADING BUGS THEN INCREASE THE DELAY TIME\n" // // IF you increase the time and still got no loading then it maybe wont work dude :)
          __src.push(FINAL)

          if (_._.indexOf(src) >= 0) {
          //console.log('src found in _._');
          var replace = "data-" + random2 + "=\"" + random3 + "\""
          _._ = _._.replace(src,replace)
          } // if (_._.indexOf(src) >= 0) {
          //******************************//

           } //  if if (src.indexOf(".js") >= 0) {
           else{

          //******************************//
         console.log('enter_src')
         var src_onlyurl = src.replace(new RegExp(/^ /gmi),"");
             src_onlyurl = src_onlyurl.replace(new RegExp(/src[=]["]/gmi),"");
             src_onlyurl = src_onlyurl.replace(new RegExp(/["]/gmi),"");


         var random = randomize('Aa', 100)
         var random2 = randomize('Aa', 100)
         var random3 = randomize('Aa', 100)
         var FINAL = "\nvar " + random + " = document.querySelectorAll('[data-" + random2 + "=\"" + random3 + "\"]');\n$('[data-" + random2 + "=\"" + random3 + "\"]').attr(\"src\", \"" + src_onlyurl + "\");\n"
          __src.push(FINAL)

            if (_._.indexOf(src) >= 0) {
            //console.log('src found in _._');
            var replace = "data-" + random2 + "=\"" + random3 + "\""
            _._ = _._.replace(src,replace)
            } // if (_._.indexOf(src) >= 0) {
            //******************************//

        


            } // else from if (src.indexOf(".jpg") >= 0) {

           } // else from if (src.indexOf("<?php") >= 0) {

          } // else from if (src.indexOf("jquery.min.js") >= 0) {

         } // else from if (src.indexOf("____________________________________________________________________.js") >= 0) {	

        }); // regex.forEach(function(val, key) {




























         /*  content  */

        /*
         regex_content.forEach(function(content, key) {

          if (content.indexOf("<?php") >= 0) {
          // ..
          }
          else{

         console.log('enter_content')
         var content_onlyurl = content.replace(new RegExp(/^ /gmi),"");
             content_onlyurl = content_onlyurl.replace(new RegExp(/content[=]["]/gmi),"");
             content_onlyurl = content_onlyurl.replace(new RegExp(/["]/gmi),"");
             console.log('content_onlyurl:', content_onlyurl)


         var random = randomize('Aa', 100)
         var random2 = randomize('Aa', 100)
         var random3 = randomize('Aa', 100)
         var FINAL = "\nvar " + random + " = document.querySelectorAll('[data-" + random2 + "=\"" + random3 + "\"]');\n$('[data-" + random2 + "=\"" + random3 + "\"]').attr(\"content\", \"" + content_onlyurl + "\");\n"
         __content.push(FINAL)

          if (_._.indexOf(content) >= 0) {
          //console.log('content found in _._');
          var replace = "data-" + random2 + "=\"" + random3 + "\""
          _._ = _._.replace(content,replace)
          } // if (_._.indexOf(content) >= 0) {





          } // else from  if (content.indexOf("<?php") >= 0) {

          }); */




         // regex_content.forEach(function(content, key) {
          console.log('Bug avoid ! content after foreach loop : ', __content)
































         /*  id  */
         regex_id.forEach(function(id, key) {

          if (id.indexOf("<?php") >= 0) {
          // ..
          }
          else{

         //******************************//
         console.log('enter_id')
         var id_onlyurl = id.replace(new RegExp(/^ /gmi),"");
             id_onlyurl = id_onlyurl.replace(new RegExp(/id[=]["]/gmi),"");
             id_onlyurl = id_onlyurl.replace(new RegExp(/["]/gmi),"");


         var random = randomize('Aa', 100)
         var random2 = randomize('Aa', 100)
         var random3 = randomize('Aa', 100)
         var FINAL = "\nvar " + random + " = document.querySelectorAll('[data-" + random2 + "=\"" + random3 + "\"]');\n$('[data-" + random2 + "=\"" + random3 + "\"]').attr(\"id\", \"" + id_onlyurl + "\");\n"
          __id.push(FINAL)

          if (_._.indexOf(id) >= 0) {
          //console.log('id found in _._');
          var replace = "data-" + random2 + "=\"" + random3 + "\""
          _._ = _._.replace(id,replace)
          } // if (_._.indexOf(id) >= 0) {
          //******************************//




          } // else from  if (id.indexOf("<?php") >= 0) {

          }); // regex_id.forEach(function(id, key) {
          console.log('id after foreach loop : ', __id)




























         /*  class  */



         // regex_classes.forEach(function(classes, key) {
          console.log('class after foreach loop : ', __class)






// THIS SECTION WILL BE RUN AFTER THE LOOP THAT CREATE THE FAKE JS FILES
//////// Adjust the amount of fake script includes that will be generated with importedlooplength - look at your const code for easy change
loop(importedlooplength, each, function (error) {
console.log('loop for fake scripts done, error? ', error)



//  ███████╗██╗**██╗██████╗**██████╗*██████╗*████████╗
//  ██╔════╝╚██╗██╔╝██╔══██╗██╔═══██╗██╔══██╗╚══██╔══╝
//  █████╗***╚███╔╝*██████╔╝██║***██║██████╔╝***██║***
//  ██╔══╝***██╔██╗*██╔═══╝*██║***██║██╔══██╗***██║***
//  ███████╗██╔╝*██╗██║*****╚██████╔╝██║**██║***██║***
//  ╚══════╝╚═╝**╚═╝╚═╝******╚═════╝*╚═╝**╚═╝***╚═╝***


                _.string = __.join("\n")
                console.log('_.string: ' + _.string);

                _.string_src = __src.join("\n")
                console.log('_.string_src: ' + _.string_src);

                _.string_content = __content.join("\n")
                console.log('_.string_content: ' + _.string_content);

                _.string_id = __id.join("\n")
                console.log('_.string_id: ' + _.string_id);

                _.string_class = __class.join("\n")
                console.log('_.string_class: ' + _.string_class);

                             // remove this here to remove the credits from www.hornyfamily.online but only inside the ''
                _.combined = '\n//  *██████╗██████╗*███████╗*█████╗*████████╗███████╗██████╗*****██████╗*██╗***██╗*******************************************************************************************\n//  ██╔════╝██╔══██╗██╔════╝██╔══██╗╚══██╔══╝██╔════╝██╔══██╗****██╔══██╗╚██╗*██╔╝*******************************************************************************************\n//  ██║*****██████╔╝█████╗**███████║***██║***█████╗**██║**██║****██████╔╝*╚████╔╝********************************************************************************************\n//  ██║*****██╔══██╗██╔══╝**██╔══██║***██║***██╔══╝**██║**██║****██╔══██╗**╚██╔╝*********************************************************************************************\n//  ╚██████╗██║**██║███████╗██║**██║***██║***███████╗██████╔╝****██████╔╝***██║**********************************************************************************************\n//  *╚═════╝╚═╝**╚═╝╚══════╝╚═╝**╚═╝***╚═╝***╚══════╝╚═════╝*****╚═════╝****╚═╝**********************************************************************************************\n//  *************************************************************************************************************************************************************************\n//  ██╗****██╗██╗****██╗██╗****██╗***██╗**██╗*██████╗*██████╗*███╗***██╗██╗***██╗███████╗*█████╗*███╗***███╗██╗██╗**██╗***██╗*██████╗*███╗***██╗██╗*****██╗███╗***██╗███████╗\n//  ██║****██║██║****██║██║****██║***██║**██║██╔═══██╗██╔══██╗████╗**██║╚██╗*██╔╝██╔════╝██╔══██╗████╗*████║██║██║**╚██╗*██╔╝██╔═══██╗████╗**██║██║*****██║████╗**██║██╔════╝\n//  ██║*█╗*██║██║*█╗*██║██║*█╗*██║***███████║██║***██║██████╔╝██╔██╗*██║*╚████╔╝*█████╗**███████║██╔████╔██║██║██║***╚████╔╝*██║***██║██╔██╗*██║██║*****██║██╔██╗*██║█████╗**\n//  ██║███╗██║██║███╗██║██║███╗██║***██╔══██║██║***██║██╔══██╗██║╚██╗██║**╚██╔╝**██╔══╝**██╔══██║██║╚██╔╝██║██║██║****╚██╔╝**██║***██║██║╚██╗██║██║*****██║██║╚██╗██║██╔══╝**\n//  ╚███╔███╔╝╚███╔███╔╝╚███╔███╔╝██╗██║**██║╚██████╔╝██║**██║██║*╚████║***██║***██║*****██║**██║██║*╚═╝*██║██║███████╗██║██╗╚██████╔╝██║*╚████║███████╗██║██║*╚████║███████╗\n//  *╚══╝╚══╝**╚══╝╚══╝**╚══╝╚══╝*╚═╝╚═╝**╚═╝*╚═════╝*╚═╝**╚═╝╚═╝**╚═══╝***╚═╝***╚═╝*****╚═╝**╚═╝╚═╝*****╚═╝╚═╝╚══════╝╚═╝╚═╝*╚═════╝*╚═╝**╚═══╝╚══════╝╚═╝╚═╝**╚═══╝╚══════╝\n//  *************************************************************************************************************************************************************************\n\n/*-------------------- WWW.HORNYFAMILY.ONLINE --------------------*/'
                + '\n\n\n\n\n\n\n\n\n$(document).ready(function() {'
                + '\n\n\n\n /*HREF*/\n' + _.string 
                + '\n\n\n\n /*SCRIPTS(SRC)*/\n' + _.string_src
                + '\n\n\n\n /*CONTENT*/\n' + _.string_content
                + '\n\n\n\n /*ID*/\n' + _.string_id
                + '\n\n\n\n /*CLASS*/\n' + _.string_class
                + '\n\n}); /*  $(document).ready(function() {   */'


                //  *****██╗███████╗****██╗███╗***██╗*██████╗██╗*****██╗***██╗██████╗*███████╗
                //  *****██║██╔════╝****██║████╗**██║██╔════╝██║*****██║***██║██╔══██╗██╔════╝
                //  *****██║███████╗****██║██╔██╗*██║██║*****██║*****██║***██║██║**██║█████╗**
                //  ██***██║╚════██║****██║██║╚██╗██║██║*****██║*****██║***██║██║**██║██╔══╝**
                //  ╚█████╔╝███████║****██║██║*╚████║╚██████╗███████╗╚██████╔╝██████╔╝███████╗
                //  *╚════╝*╚══════╝****╚═╝╚═╝**╚═══╝*╚═════╝╚══════╝*╚═════╝*╚═════╝*╚══════╝
                //  **************************************************************************
                // you need to include the file or the code inside the file into your website
                // like every other srcipt you import as example
                // <srcipt src="catalog/view/newfile.js" type="text/javasrcipt"></srcipt>
                fs.appendFile(crypted_file4, _.combined, function (err) {
                if (err) {
                console.log('WARNING - run into error at saving ' + crypted_file4 + ' Maybe try again :)', err);
                } // if (err) {
                else {
                console.log('############### FINISH - Please look for ' + crypted_file4 + ' in your root folder where you run this script from ####################')
                }; // else from if (err) {
                }); // fs.appendFile('_INCLUDE-THIS-CODE_EXPORT.txt', _.string, function (err) {


               



                //  *██████╗██████╗*██╗***██╗██████╗*████████╗███████╗██████╗*****███████╗██╗██╗*****███████╗
                //  ██╔════╝██╔══██╗╚██╗*██╔╝██╔══██╗╚══██╔══╝██╔════╝██╔══██╗****██╔════╝██║██║*****██╔════╝
                //  ██║*****██████╔╝*╚████╔╝*██████╔╝***██║***█████╗**██║**██║****█████╗**██║██║*****█████╗**
                //  ██║*****██╔══██╗**╚██╔╝**██╔═══╝****██║***██╔══╝**██║**██║****██╔══╝**██║██║*****██╔══╝**
                //  ╚██████╗██║**██║***██║***██║********██║***███████╗██████╔╝****██║*****██║███████╗███████╗
                //  *╚═════╝╚═╝**╚═╝***╚═╝***╚═╝********╚═╝***╚══════╝╚═════╝*****╚═╝*****╚═╝╚══════╝╚══════╝
                //  Here the original file will exported as crypted file
                 
                
                   



                _._ = _._
                // remove everything inside "" to remove the credits of www.hornyfamily.online
                + "\n<!--  *██████╗██████╗*███████╗*█████╗*████████╗███████╗██████╗*****██████╗*██╗***██╗*******************************************************************************************-->\n<!--  ██╔════╝██╔══██╗██╔════╝██╔══██╗╚══██╔══╝██╔════╝██╔══██╗****██╔══██╗╚██╗*██╔╝*******************************************************************************************-->\n<!--  ██║*****██████╔╝█████╗**███████║***██║***█████╗**██║**██║****██████╔╝*╚████╔╝********************************************************************************************-->\n<!--  ██║*****██╔══██╗██╔══╝**██╔══██║***██║***██╔══╝**██║**██║****██╔══██╗**╚██╔╝*********************************************************************************************-->\n<!--  ╚██████╗██║**██║███████╗██║**██║***██║***███████╗██████╔╝****██████╔╝***██║**********************************************************************************************-->\n<!--  *╚═════╝╚═╝**╚═╝╚══════╝╚═╝**╚═╝***╚═╝***╚══════╝╚═════╝*****╚═════╝****╚═╝**********************************************************************************************-->\n<!--  *************************************************************************************************************************************************************************-->\n<!--  ██╗****██╗██╗****██╗██╗****██╗***██╗**██╗*██████╗*██████╗*███╗***██╗██╗***██╗███████╗*█████╗*███╗***███╗██╗██╗**██╗***██╗*██████╗*███╗***██╗██╗*****██╗███╗***██╗███████╗-->\n<!--  ██║****██║██║****██║██║****██║***██║**██║██╔═══██╗██╔══██╗████╗**██║╚██╗*██╔╝██╔════╝██╔══██╗████╗*████║██║██║**╚██╗*██╔╝██╔═══██╗████╗**██║██║*****██║████╗**██║██╔════╝-->\n<!--  ██║*█╗*██║██║*█╗*██║██║*█╗*██║***███████║██║***██║██████╔╝██╔██╗*██║*╚████╔╝*█████╗**███████║██╔████╔██║██║██║***╚████╔╝*██║***██║██╔██╗*██║██║*****██║██╔██╗*██║█████╗**-->\n<!--  ██║███╗██║██║███╗██║██║███╗██║***██╔══██║██║***██║██╔══██╗██║╚██╗██║**╚██╔╝**██╔══╝**██╔══██║██║╚██╔╝██║██║██║****╚██╔╝**██║***██║██║╚██╗██║██║*****██║██║╚██╗██║██╔══╝**-->\n<!--  ╚███╔███╔╝╚███╔███╔╝╚███╔███╔╝██╗██║**██║╚██████╔╝██║**██║██║*╚████║***██║***██║*****██║**██║██║*╚═╝*██║██║███████╗██║██╗╚██████╔╝██║*╚████║███████╗██║██║*╚████║███████╗-->\n<!--  *╚══╝╚══╝**╚══╝╚══╝**╚══╝╚══╝*╚═╝╚═╝**╚═╝*╚═════╝*╚═╝**╚═╝╚═╝**╚═══╝***╚═╝***╚═╝*****╚═╝**╚═╝╚═╝*****╚═╝╚═╝╚══════╝╚═╝╚═╝*╚═════╝*╚═╝**╚═══╝╚══════╝╚═╝╚═╝**╚═══╝╚══════╝-->\n<!--  *************************************************************************************************************************************************************************-->\n\n<!-------------------- WWW.HORNYFAMILY.ONLINE -------------------->\n\n\n\n"

                fs.appendFile(crypted_file2, _._, function (err) {
                if (err) {
                console.log('WARNING - run into error at saving crypted file', err);
                } // if (err) {
                else {
                console.log('############### FINISH - Please look for ' + crypted_file2 + ' in your root folder ####################')
                }; // else from if (err) {
                }); // fs.appendFile('_INCLUDE-THIS-CODE_EXPORT.txt', _.string, function (err) {

        




                      }); // loop(importedlooplength, each, function (error) {

                      function each (next, i) {
                      setTimeout(function () {

                      _.fakescripts = randomize('Aa0', 100)
                      _.fakescripts_empty = randomize('Aa0', 100)
                      _.fakescripts_empty2 = randomize('Aa0', 100)
                      _._ = _._
                      // real fake scripts
                      + "\n<script src=\"_/" + _.fakescripts + ".js\"></script>\n"


                      
                         var fakenumber = randomize('000')
                         loop(fakenumber, each, function (error) {
                         console.log('loop for fake scripts done, error? ', error)
                         // ..
                         }); // loop(importedlooplength, each, function (error) {





                            function each (next, i) {
                            setTimeout(function () {


                            var fakegenerator = randomize('Aa0!', 1000);
                            var fakegenerator2 = randomize('Aa0!', 690);

                            var createme = "var " + randomize('Aa', 1000) + " = " + fakegenerator2
                            var createme2 = createme + "\n\nvar " + randomize('Aa', 1000) + " = " + fakegenerator2

                            var exportrdy = "./FAKE_SCRIPTS_EXPORT_AREA/" + _.fakescripts + ".js"

                                fs.appendFile(exportrdy, createme2, function (err) {
                                if (err) {
                                console.log('WARNING - run into error at saving files into /FAKE_SCRIPTS_EXPORT_AREA/', err);
                                } // if (err) {
                                else {
                                console.log('############### FINISH CURRENT LOOP FOR - ' + exportrdy + ' - ####################')
                                }; // else from if (err) {
                                }); // fs.appendFile('_INCLUDE-THIS-CODE_EXPORT.txt', _.string, function (err) {

                                }, 50) //  setTimeout(function () {
                                } // function each (next, i) {






                      next()

                      }, 50) //  setTimeout(function () {
                      } // function each (next, i) {






    } // if (!err) {
    else {
    console.log('WARNING - run into error at reading imported file: ', err);
    } // else from  if (!err) { 
}); // fs.readFile(filePath, {encoding: 'utf-8'}, function(err,data){














// created by - www.hornyfamily.online
} // exports.printMsg = function() {
